ambry-clustermap/src/test/java/com/github/ambry/clustermap/HelixHealthReportAggregationTaskTest.java

```diff
@@ -0,0 +1,252 @@
+import com.github.ambry.utils.TestUtils;
+import com.github.ambry.utils.Time;
+import java.io.File;
+import java.io.IOException;
+import java.net.BindException;
+import java.nio.file.Files;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Properties;
+import java.util.Random;
+import org.apache.helix.AccessOption;
+import org.apache.helix.HelixDataAccessor;
+import org.apache.helix.HelixProperty;
+import org.apache.helix.InstanceType;
+import org.apache.helix.PropertyKey;
+import org.apache.helix.model.InstanceConfig;
+import org.apache.helix.zookeeper.datamodel.ZNRecord;
+import org.apache.zookeeper.data.Stat;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Assume;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+
+/**
+ * Unit test class for {@link HelixHealthReportAggregatorTask}.
+ */
+@RunWith(Parameterized.class)
+public class HelixHealthReportAggregationTaskTest {
+  private static final String CLUSTER_NAME = "Ambry-prod";
+  private static final String INSTANCE_NAME = "helix.ambry.com";
+  private static final long RELEVANT_PERIOD_IN_MINUTES = 60;
+  private static final String HEALTH_REPORT_NAME_ACCOUNT = "AccountReport";
+  private static final String STATS_FIELD_NAME_ACCOUNT = "AccountStats";
+  private static final String HEALTH_REPORT_NAME_PARTITION = "PartitionClassReport";
+  private static final String STATS_FIELD_NAME_PARTITION = "PartitionClassStats";
+  private MockHelixManager mockHelixManager;
+  private HelixHealthReportAggregatorTask task;
+  private MockTime mockTime;
+  private MockHelixAdmin mockHelixAdmin;
+  private ClusterMapConfig clusterMapConfig;
+  private TestUtils.ZkInfo zkInfo;
+  private final ObjectMapper mapper = new ObjectMapper();
+
+  private final boolean enableAggregatedMonthlyAccountReport;
+
+  @Parameterized.Parameters
+  public static List<Object[]> data() {
+    return Arrays.asList(new Object[][]{{false}, {true}});
+  }
+
+  public HelixHealthReportAggregationTaskTest(boolean enableAggregatedMonthlyAccountReport) throws IOException {
+    this.enableAggregatedMonthlyAccountReport = enableAggregatedMonthlyAccountReport;
+    File tempDir = Files.createTempDirectory("HelixHealthReportAggregationTask-" + new Random().nextInt(1000)).toFile();
+    String tempDirPath = tempDir.getAbsolutePath();
+    int zkPort = 13188;
+    int maxRetries = 1000;
+    int numRetries = 0;
+    while (true) {
+      if (numRetries > maxRetries) {
+        throw new IOException("No ports are available to start zookeeper server");
+      }
+      try {
+        zkInfo = new TestUtils.ZkInfo(tempDirPath, "DC1", (byte) 0, zkPort, true);
+        break;
+      } catch (Exception e) {
+        if (e instanceof BindException || e.getCause() instanceof BindException) {
```

Comment:
It doesn't seem to throw `BindException` directly. We probably can remove `e instanceof BindException`